2020-04-introducing-mdjs-live-demos-everywhere/README.md

```diff
@@ -0,0 +1,360 @@
+---
+title: Introducing mdjs - live demos everywhere
+published: false
+description: Combine markdown and javascript to create the ultimate documentation to allow live demos everywhere.
+tags: markdown, javascript, webcomponents, demos
+cover_image: https://github.com/open-wc/blog-posts/blob/2019-11-storybook-for-web-components/2019-11-storybook-for-web-components-on-steroids/images/nong-vang-9pw4TKvT3po-unsplash.jpg?raw=true
+---
+
+Every shared code needs had written documentation to show what it can be used for and what the idea behind is.
+Users should at least on the high level understand what the are using for what and why.
+
+On the web, we have many many different ways of writing documentation.
+However, one thing almost all of them have in common is that they rely on [Markdown](https://en.wikipedia.org/wiki/Markdown) or of variations of it.
+
+It is supported literally everywhere (vscode, atom, github, gitlab, dev.to, npmjs, ...)
+
+## For tools that do not run in the browser
+
+In that case, you will mostly share code snippets that people will need to run in their own projects so traditional static site generators like [Docusaurus](https://docusaurus.io/), [VuePress](https://vuepress.vuejs.org/), [Gatsby](https://www.gatsbyjs.org/), ... work great. All of them have full support for Markdown and therefore can create beautiful documentation pages with code snippets/highlighting, ...
+
+Honestly, if that is your use case almost everything should be possible as long as you feel comfortable with the tools ecosystem/framework.
+
+## For (visual) components that do run in the browser
+
+Here people do expect a live demo to see different options in action. So pure Markdown is usually not enough as we now want to actually execute code and "insert" our working component. This requires specialized handling for each framework.
```

Comment:
```suggestion
In this context, users probably do expect a live demo to see all the different options of your component in action. So pure Markdown is usually not enough as we now want to actually execute code and "insert" our working component in our documentation somehow. This would require specialized handling for each framework.
```